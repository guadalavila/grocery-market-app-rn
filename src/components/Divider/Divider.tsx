import React from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { Colors } from '../../utils/Colors';
import { Sizes } from '../../utils/GlobalStyles';

interface IDividerProps {
    color?: string;
    height?: number;
    style?: StyleProp<ViewStyle>;
    marginVertical?: number;
    marginHorizontal?: number;
}

const Divider = ({
    color = Colors.greyLight,
    height = 1.5,
    style = {},
    marginHorizontal = Sizes.S,
    marginVertical = Sizes.M,
}: IDividerProps) => {
    return (
        <View
            style={[
                styles.container,
                style,
                {
                    height: height,
                    backgroundColor: color,
                    marginHorizontal: marginHorizontal,
                    marginVertical: marginVertical,
                },
            ]}
        />
    );
};

export default Divider;

const styles = StyleSheet.create({
    container: {},
});
