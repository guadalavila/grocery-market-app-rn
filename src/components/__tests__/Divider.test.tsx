import { render, screen } from '@testing-library/react-native';
import React from 'react';
import Divider from '../Divider/Divider';

describe('test Divider component', () => {
    test('should render ', () => {
        const component = render(<Divider />);
        expect(component).toMatchSnapshot();
    });
    test('should render component', () => {
        render(<Divider />);
        expect(screen).toBeTruthy();
    });
});
