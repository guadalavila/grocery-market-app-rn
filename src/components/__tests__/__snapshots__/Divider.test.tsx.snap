// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Divider component should render  1`] = `
<View
  style={
    Array [
      Object {},
      Object {},
      Object {
        "backgroundColor": "#C5C5C5",
        "height": 1.5,
        "marginHorizontal": 8,
        "marginVertical": 16,
      },
    ]
  }
/>
`;
